sharktank/sharktank/utils/patching.py

```diff
@@ -50,30 +66,98 @@ class SaveModuleResultTensorsPatch(Patch):
     Users must call finalize() once all tensors have been accumulated.
     """
 
-    def __init__(self):
-        self.tensors = {}
+    def __init__(self, with_before_forward: bool = False):
+        self.with_before_forward = with_before_forward
+        self.tensors: dict[str, torch.Tensor] = {}
         # Map of module_name to last used index for duplicated tensors.
-        self.duplicate_tensors = {}
-
-    def after_forward(self, module_name: str, module: torch.nn.Module, results):
-        if not isinstance(results, torch.Tensor):
+        self.duplicate_tensors = dict[str, torch.Tensor] = {}
```

Comment:
This is a bug and was caught by the CI for this PR: https://github.com/nod-ai/shark-ai/actions/runs/15046056698/job/42288759451

Should be `self.duplicate_tensors : dict[str, torch.Tensor] = {}` (First `=` should be `:`).

Make sure to check why the CIs fail. I know the data one in particular is flaky, but in this case it did run.